### PR TITLE
Update for Wagtail 6.3

### PR DESCRIPTION
**Pull request by nickmoreton**
This pull request includes updates to the CI configuration and project metadata to support newer versions of Python, Django, and Wagtail.

Updates to CI configuration:

* [`.github/workflows/ci.yml`](diffhunk://#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fL16-R22): Updated the matrix to include Python 3.12 and removed Python 3.8. Updated Wagtail versions to include 6.2 and added a specific combination for Python 3.12, Django 5.1, and Wagtail 6.3.

Updates to project metadata:

* [`pyproject.toml`](diffhunk://#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711L20-R22): Updated the classifiers to include Django 5.0 and 5.1, and removed Django 3.2 and 4.1.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -13,14 +13,18 @@ jobs:
...
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
         django-version: ["4.2", "5.0"]
-        wagtail-version: ["5.2", "6.0", "6.1"]
+        wagtail-version: ["5.2", "6.2"]
+        include:
+          - python-version: "3.12"
+            django-version: "5.1"
+            wagtail-version: "6.3"
```
Comment (RealOrangeOne) [inline .github/workflows/ci.yml:22]: Question: Why is Wagtail 6.3 only being tested on very specific Python and Django versions?

Comment (nickmoreton) [reply .github/workflows/ci.yml:22]: Oh yeah I could include 3.10 and 3.11 as well. I'll do an update.

Comment (nickmoreton) [reply .github/workflows/ci.yml]: I've switched it over to use exclude, thanks

**Comment by nickmoreton** [comment]
Thanks

**Pull request merged**